sycl/source/builtins_generator.py

```diff
@@ -1052,13 +1052,13 @@ def custom_nan_invoke(return_type, arg_types, arg_names):
                              RelDef("bool", ["sgenfloat"], invoke_name="SignBitSet"),
                              RelDef("boolelements0", ["mgenfloat"])],
                  "any": [Def("int", ["vigeninteger"], custom_invoke=get_custom_any_all_vec_invoke("Any")),
-                         Def("bool", ["sigeninteger"], custom_invoke=(lambda return_type, arg_types, arg_names: f'  return bool(int(detail::msbIsSet({arg_names[0]})));')),
+                         Def("bool", ["sigeninteger"], template_scalar_args=True, custom_invoke=(lambda return_type, arg_types, arg_names: f'  return bool(int(detail::msbIsSet({arg_names[0]})));')),
```

Comment:
Is this marked as deprecated?

Reply:
This variant is not deprecated. According to the specification integer scalar types should return a bool value.

Reply:
![image](https://github.com/intel/llvm/assets/55163882/2133e630-2b9e-4a80-9c51-45c626cadc9c)

> Overload (2):
>
> This overload is deprecated in SYCL 2020.
>
> Constraints: Available only if GenInt is one of the following types:
>
> signed char
> short
> int
> long
> long long
> marray<signed char, N>
> marray<short, N>
> marray<int, N>
> marray<long, N>
> marray<long long, N>
>
> Returns: When x is a scalar, returns a Boolean telling whether the most
> significant bit of x is set. When x is marray, returns a Boolean telling
> whether the most significant bit of any element in x is set

Reply:
Oh, good point. No it has not been deprecated yet.